test/unit/metrics/test_metrics_engine.py

```diff
@@ -188,13 +200,13 @@ def negative_results(y_true, y_pred):
 
         result = metrics.metric_by_group(negative_results, y_a, y_p, gid)
 
-        assert result.overall == -16
-        assert result.by_group[0] == -10
-        assert result.by_group[1] == -6
-        assert result.minimum == -10
-        assert result.maximum == -6
-        assert result.range == 4
-        assert np.isnan(result.range_ratio)
+        assert result['overall'] == -16
+        assert result['group_0'] == -10
+        assert result['group_1'] == -6
+        assert result['min'] == -10
+        assert result['max'] == -6
+        assert result['range'] == 4
+        assert np.isnan(result['range_ratio'])
```

Comment:
if the result is a `Bunch`, you could still access them as `result.min` etc.

Reply:
I decided against using that.

Reply:
also, for some of these you could keep the old _key_ name, to keep it more backward compatible.

Reply:
> I decided against using that.

care to elaborate why?

Reply:
Thinking it over, I wasn't keen on a dependency on undocumented behaviour.

Reply:
But we talked about it and the conclusion was to document it. That's why I've created  https://github.com/scikit-learn/scikit-learn/pull/16404

Also, if it's supposed to not be used, you could use a `dict` instead of a `Bunch`.

Reply:
I _am_ using a `dict`.